src/components/CourseList/CourseList.tsx

```diff
@@ -0,0 +1,50 @@
+import { useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { useAppSelector } from '../../app/hooks';
+import { Courses } from '../../types/Courses';
+import { CourseCard } from '../CourseCard';
+import { Pagination } from '../Pagination';
+import './CourseList.scss';
+
+export const CourseList: React.FC = () => {
+  const { courses } = useAppSelector(state => state.courses);
+
+  const [searchParams] = useSearchParams();
+
+  const currentPage = searchParams.get('page') || '1';
+  const perPage = 10;
+
+  let leftLimit = 0;
+  let rightLimit = courses.length;
+
+  if (currentPage && perPage) {
+    leftLimit = perPage * (+currentPage - 1);
+
+    rightLimit = perPage * +currentPage > courses.length
+      ? courses.length
+      : perPage * +currentPage;
+  } else {
+    leftLimit = 0;
+    rightLimit = courses.length;
+  }
```

Comment:
1. As far as I understand, the condition in `if (currentPage && perPage)` is always true, since both currentPage and perPage cannot be falsy values. So we never get into the **else** statement and there is no need for an **if...else** here

2. It seems to me that in this case it is better to use useEffect to set values for limits. This will help to avoid unnecessary reassignments of variables. In the first video, I just added console.log inside your **if (currentPage && perPage)** statement:

https://user-images.githubusercontent.com/72460025/232226554-551964b0-6483-4e65-b288-bdd311821135.mp4


and in the second video i added console.log inside useEffect in the solution i will suggest you below


https://user-images.githubusercontent.com/72460025/232226696-6fccbe90-0637-46dc-8422-5af7274d8aa3.mp4

```suggestion
  const [leftLimit, setLeftLimit] = useState(0);
  const [rightLimit, setRightLimit] = useState(courses.length);

  const currentPage = searchParams.get('page') || '1';
  const perPage = 10;
  
  useEffect(() => {
    const maxRightLimit = perPage * +currentPage;

    if (maxRightLimit > courses.length) {
      setRightLimit(courses.length);
    } else {
      setRightLimit(maxRightLimit);
    }

    setLeftLimit(perPage * (+currentPage - 1));
  }, [currentPage, perPage, courses])
```

Reply:
Good suggestion. 👍
Done ✅